test/Microsoft.OpenAPI.OData.Reader.Tests/Generator/OpenApiResponseGeneratorTests.cs

```diff
@@ -320,5 +320,116 @@ public void CreateResponseForEdmActionWhenErrorResponsesAsDefaultIsSet(string ac
                 Assert.Equal(new string[] { responseCode, "4XX", "5XX" }, responses.Select(r => r.Key));
             }
         }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void CreateResponseForDeltaEdmFunctionReturnCorrectResponses(bool enableOdataAnnotationRef)
+        {
+            // Arrange
+            IEdmModel model = EdmModelHelper.GraphBetaModel;
+            OpenApiConvertSettings settings = new()
+            {
+                EnableODataAnnotationReferencesForResponses = enableOdataAnnotationRef
+            };
+            ODataContext context = new(model, settings);
+
+            // Act
+            IEdmFunction operation = model.SchemaElements.OfType<IEdmFunction>().First(o => o.Name == "delta" &&
+                   o.Parameters.First().Type.FullName() == "Collection(microsoft.graph.application)");
+            Assert.NotNull(operation); // guard
+            OpenApiResponses responses = context.CreateResponses(operation);
+            string json = responses.SerializeAsJson(OpenApiSpecVersion.OpenApi3_0);
+
+            // Assert
+            Assert.NotNull(responses);
+            Assert.NotEmpty(responses);
+            if (enableOdataAnnotationRef)
+            {
+                Assert.Equal(@"{
+  ""200"": {
+    ""description"": ""Success"",
+    ""content"": {
+      ""application/json"": {
+        ""schema"": {
+          ""title"": ""Collection of application"",
+          ""type"": ""object"",
+          ""allOf"": [
+            {
+              ""$ref"": ""#/components/schemas/BaseDeltaFunctionResponse""
+            },
+            {
+              ""type"": ""object"",
+              ""properties"": {
+                ""value"": {
+                  ""type"": ""array"",
+                  ""items"": {
+                    ""anyOf"": [
+                      {
+                        ""$ref"": ""#/components/schemas/microsoft.graph.application""
+                      },
+                      {
+                        ""type"": ""object"",
+                        ""nullable"": true
```

Comment:
shouldn't this be a oneOf?

Reply:
Yes, ideally. See the discussion we had [here](https://github.com/microsoft/OpenAPI.NET.OData/issues/190#issuecomment-1177693486) on why we are currently using `anyOf`.

Reply:
right, I forgot, thanks for the reminder, resolving.